plugins/workload-management/src/main/java/org/opensearch/plugin/wlm/action/CreateQueryGroupRequest.java

```diff
@@ -0,0 +1,243 @@
+                fieldName = parser.currentName();
+            } else if (token.isValue()) {
+                if (fieldName.equals("name")) {
+                    name = parser.text();
+                } else if (fieldName.equals("resiliency_mode")) {
+                    mode = ResiliencyMode.fromName(parser.text());
+                } else {
+                    throw new XContentParseException("unrecognised [field=" + fieldName + " in QueryGroup");
+                }
+            } else if (token == XContentParser.Token.START_OBJECT) {
+                if (!fieldName.equals("resourceLimits")) {
+                    throw new XContentParseException("Invalid field passed. QueryGroup does not support " + fieldName + ".");
+                }
+                while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
+                    if (token == XContentParser.Token.FIELD_NAME) {
+                        fieldName = parser.currentName();
+                    } else {
+                        resourceLimits.put(ResourceType.fromName(fieldName), parser.doubleValue());
+                    }
+                }
+            }
+        }
+        return new CreateQueryGroupRequest(name, UUIDs.randomBase64UUID(), mode, resourceLimits, Instant.now().getMillis());
+    }
+
+    @Override
+    public ActionRequestValidationException validate() {
+        return null;
+    }
+
+    /**
+     * name getter
+     */
+    public String getName() {
+        return name;
+    }
+
+    /**
+     * name setter
+     * @param name - name to be set
+     */
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    /**
+     * resourceLimits getter
+     */
+    public Map<ResourceType, Object> getResourceLimits() {
+        return resourceLimits;
+    }
+
+    /**
+     * resourceLimits setter
+     * @param resourceLimits - resourceLimit to be set
+     */
+    public void setResourceLimits(Map<ResourceType, Object> resourceLimits) {
+        this.resourceLimits = resourceLimits;
+    }
+
+    /**
+     * mode getter
+     */
+    public ResiliencyMode getResiliencyMode() {
+        return resiliencyMode;
+    }
+
+    /**
+     * mode setter
+     * @param resiliencyMode - mode to be set
+     */
+    public void setResiliencyMode(ResiliencyMode resiliencyMode) {
+        this.resiliencyMode = resiliencyMode;
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        super.writeTo(out);
+        out.writeString(name);
+        out.writeString(_id);
+        out.writeString(resiliencyMode.getName());
+        out.writeMap(resourceLimits, ResourceType::writeTo, StreamOutput::writeGenericValue);
+        out.writeLong(updatedAtInMillis);
+    }
+
+    /**
+     * _id getter
+     */
+    public String get_id() {
+        return _id;
+    }
+
+    /**
+     * UUID setter
+     * @param _id - _id to be set
+     */
+    public void set_id(String _id) {
+        this._id = _id;
+    }
```

Comment:
Get rid of all the setter methods. `CreateQueryGroupRequest` once created should be immutable